Compilation error on ESP32

I'm building this for ESP32 using platformio.ini:

```ini
[env:lolin32]
platform = espressif32
board = lolin32
framework = arduino
monitor_speed = 9600

lib_deps =
  MD_YX5300
```

Results in this error:

```
.pio/libdeps/lolin32/MD_YX5300_ID5592/src/MD_YX5300.h:12:28: fatal error: SoftwareSerial.h: No such file or directory
```

So I add `SoftwareSerial` to `lib_deps` in `platformio.ini` but the error doesn't go away. So I found a esp32 version of SoftwareSerial: https://github.com/plerup/espsoftwareserial

After adding `EspSoftwareSerial` to `lib_deps` in `platformio.ini` it compiles, but now throws many errors like this:

```
.pio/libdeps/lolin32/MD_YX5300_ID5592/src/MD_YX5300.h:1125:12: error: 'MD_YX5300::_timeout' will be initialized after [-Werror=reorder]
   uint32_t _timeout;  ///< timeout for return serial message
            ^
.pio/libdeps/lolin32/MD_YX5300_ID5592/src/MD_YX5300.h:1124:8: error:   'bool MD_YX5300::_synch' [-Werror=reorder]
   bool _synch;        ///< synchronous (wait for response) if true
        ^
.pio/libdeps/lolin32/MD_YX5300_ID5592/src/MD_YX5300.h:461:3: error:   when initialized here [-Werror=reorder]
   MD_YX5300(uint8_t pinRx, uint8_t pinTx) : 
```

To fix that I added this to `platformio.ini`:

```ini
; Used for YX5300 audio player, see https://github.com/MajicDesigns/MD_YX5300/issues/8
build_unflags = -Werror=reorder
```

And it now compiles. But is this the correct way to do it @MajicDesigns?

--- a/src/MD_YX5300.h
+++ b/src/MD_YX5300.h
@@ -462,7 +462,7 @@ public:
 #if USE_SOFTWARESERIAL
     _Serial(pinRx, pinTx),
 #endif
-  _timeout(1000), _synch(true), _cbStatus(nullptr)
+    _cbStatus(nullptr), _synch(true), _timeout(1000)
     {};
 
  /**

